intro_calculating_cred_grain.md

```diff
@@ -0,0 +1,162 @@
+
+#### Incorporate node weights
+
+There’s also a cleverly simple way to account for the existence of node weights. One requirement for the existence of a stationary distribution that was skipped is the concept of ergodicity, wherein every state must be reachable from any other state, given a long enough probabilistic walk across the graph’s edges†. To achieve ergodicity, SourceCred uses a "teleportation" system by adding a probability, ⍺, of teleporting to a random location in the graph. SourceCred currently implements this with a seed node, which all nodes teleporting to with likelihood ⍺. From the seed node, the probability of landing on a given node in the graph is then given by the node weights. If we use _A_ to describe the weights of each node in a vector form, then our old transition matrix <img src="https://render.githubusercontent.com/render/math?math=P_{uv}"> becomes the new transition matrix <img src="https://render.githubusercontent.com/render/math?math=(1-\alpha)P_{uv} + \alpha A">. The stationary distribution is then the vector <img src="https://render.githubusercontent.com/render/math?math=\vec v"> such that <img src="https://render.githubusercontent.com/render/math?math=\left((1-\alpha)P_{uv} + \alpha A\right)\vec v = \vec v">.
+
+#### The origin of Cred
+
+In this way, Cred originates at the weighted nodes and flows out to actions from there. Interestingly, this is mathematically equivalent to adding edges to each and every node, with edge-weight proportional to ⍺ times the node weight. We can interpret the necessity of teleportation in the following way: because some contributions may receive but flow no cred, there need to be sources of Cred to calculate an equilibrium value distribution, which requires some contributions to mint cred.
+
+#### Cred-minting philosophy
+
+As a matter of design philosophy, we believe cred should be minted at moments where there is review of an action, rather than the action itself. This reduces the danger of spam and reinforces the idea that contributors dictate what is valuable. For other projects adopting SourceCred, contributions that mint cred could also be actions performed by a moderator or maintainer.
+
+#### Summary up to this point:
+
+In this way, PageRank takes weights in and outputs scores that value contributions based on their relationship to each other. The eigenvector <img src="https://render.githubusercontent.com/render/math?math=\vec v"> then consists of the Cred scores for the various nodes, including the contributors, so we’ve accomplished our goal!
+
+## Cred over time
+
+#### Time-based Cred constraints
+
+SourceCred has additionally innovated on PageRank to accommodate properties that Cred should demonstrate over time. The most important properties in this respect are that:
+
+-   a contributor should never lose cred. They ought to become more diluted as other contributors continue to earn new cred, but they’ll still continue to earn Grain on that Cred.
```

Comment:
"A contributor should never lose cred" -- I don't agree with this. Consider the following example.

We've boosted feature X so that it's worth 100 cred. I go ahead and do initial work on feature X, and receive most of that 100 cred. It transpires that my work was buggy and unmaintainable, and so now new contributors need to come and rewrite feature. Their new edges dilute the 100 cred, so now I only get (say) 30 cred. I've now "lost" cred.

Reply:
ooh, very good point.  I've removed that claim and added the point you made.